language/gpt3/megatron/dataset.py

```diff
@@ -0,0 +1,167 @@
+import os
+import sys
+
+sys.path.append(os.environ["MEGATRON_PATH"])
+from megatron.tokenizer import build_tokenizer
+from megatron.utils import get_ltor_masks_and_position_ids
+import torch
+import argparse
+import utils
+import json
+
+from argparse import Namespace
+
+
+PROMPT_DICT = {
+    "prompt_input": (
+        "{instruction}{input}"
+    )
+}
+
+
+class Dataset:
+    def __init__(
+        self,
+        dataset_path,
+        batch_size=1,
+        args=Namespace(),
+        gen_kwards={},
+        total_count_override=None,
+        perf_count_override=None,
+        debug=False,
+    ):
+        print("Constructing QSL")
+
+        self.dataset = "cnn_dailymail"
+        self.dataset_path = dataset_path
+        self.batch_size = batch_size
+        self.debug = debug
+        self.gen_kwards = gen_kwards
+
+        ## TODO: provide arguments in command line
+        args.rank = 0
+        args.tokenizer_type = "SentencePieceTokenizer"
+        args.vocab_extra_ids = 0
+        if 'make_vocab_size_divisible_by' not in vars(args):
+            args.make_vocab_size_divisible_by = 128
+        if 'make_vocab_size_divisible_by' not in vars(args):
```

Comment:
@pgmpablo157321 typo
```suggestion
        if 'tensor_model_parallel_size' not in vars(args):
```